File: nhglib-core/src/io/github/movementspeed/nhglib/runtime/ecs/components/graphics/CameraComponent.java
package io.github.movementspeed.nhglib.runtime.ecs.components.graphics;

import com.artemis.Component;
import com.badlogic.gdx.graphics.Camera;

/**
 * Created by Fausto Napoli on 03/03/2017.
 */
public class CameraComponent extends Component {
    public Camera camera;
    public Type type;

    public enum Type {
        PERSPECTIVE,
        ORTHOGRAPHIC;

        public static CameraComponent.Type fromString(String value) {
            CameraComponent.Type type = null;

            if (value.contentEquals("perspective")) {
                type = PERSPECTIVE;
            } else if (value.contentEquals("orthographic")) {
                type = ORTHOGRAPHIC;
            }

            return type;
        }
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/core/ecs/components/graphics/CameraComponent.kt
package io.github.movementspeed.nhglib.core.ecs.components.graphics

import com.artemis.Component
import com.badlogic.gdx.graphics.Camera

/**
 * Created by Fausto Napoli on 03/03/2017.
 */
class CameraComponent : Component() {
    var camera: Camera? = null
    var type: Type? = null

    enum class Type {
        PERSPECTIVE,
        ORTHOGRAPHIC;


        companion object {

            fun fromString(value: String): CameraComponent.Type? {
                var type: CameraComponent.Type? = null

                if (value.contentEquals("perspective")) {
                    type = PERSPECTIVE
                } else if (value.contentEquals("orthographic")) {
                    type = ORTHOGRAPHIC
                }

                return type
            }
        }
    }
}
